cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/AivenApiService.java

```diff
@@ -161,6 +161,29 @@ private void createServiceAccount(
     }
   }
 
+  private String deleteServiceAccountUser(
+      String projectName, String serviceName, String serviceAccountUser) {
+    log.debug(
+        "Deleting service account user : projectName serviceName serviceAccountUser :{} {} {}",
+        projectName,
+        serviceName,
+        serviceAccountUser);
+    String uri =
+        serviceAccountApiEndpoint
+            .replace(PROJECT_NAME, projectName)
+            .replace(SERVICE_NAME, serviceName)
+            .replace("userName", serviceAccountUser);
```

Comment:
Only change I would make is to create a couple of extra statics "userName" && prinicple && resourceType aside from that this change looks good to me